Community/Tdarr_Plugin_TN10_SUBS.js

```diff
@@ -0,0 +1,193 @@
+    Description: 'This plugin extracts embedded .srt subs in one pass inside Tdarr and will optionally remove them as well as all image based subtitles. This will not extract commentary files.\n\n ',
+    Version: '1.00',
+    Link: '',
+    Tags: 'pre-processing,subtitle only,ffmpeg,configurable',
+    Inputs: [{
+        name: 'extract_lang',
+        type: 'string',
+        defaultValue: 'eng',
+        inputUI: {
+          type: 'text',
+        },
+        tooltip: 'Specify language tag(s) here for the subtitle tracks you would like to extract.\\nLeave blank to extract all subtitle tracks.\\nMust follow ISO-639-2 3 letter format. https://en.wikipedia.org/wiki/List_of_ISO_639-2_codes.\\nExample: \\neng\\nExample: \\neng,jpn'
+      },
+      {
+        name: 'remove_subs',
+        type: 'boolean',
+        defaultValue: 'false',
+        inputUI: {
+          type: 'dropdown',
+          options: [
+            'false',
+            'true',
+          ],
+        },
+        tooltip: 'Remove all subtitle streams from file after extrations.\\nThis will also remove image format subtitles that can not be extracted.'
+      },
+      {
+        name: 'overwrite_subs',
+        type: 'boolean',
+        defaultValue: 'false',
+        inputUI: {
+          type: 'dropdown',
+          options: [
+            'false',
+            'true',
+          ],
+        },
+        tooltip: 'Overwrite existing subtitle files on disk if they exist.'
+      },
+    ],
+  };
+}
+
+function plugin(file, librarySettings, inputs, otherArguments) {
+  // Must return this object at some point in the function else plugin will fail.
+  const response = {
+    processFile: true,
+    preset: '',
+    container: `.${file.container}`,
+    handBrakeMode: false,
+    FFmpegMode: true,
+    reQueueAfter: false,
+    infoLog: '',
+  };
+
+  // Check if file is a video. If it isn't then exit plugin.
+  if (file.fileMedium !== 'video') {
+    // eslint-disable-next-line no-console
+    response.infoLog += '☒File is not video \n';
+    response.processFile = false;
+    return response;
+  }
+
+  // Make sure file has subtitles.
+  let hasSubs = false;
+  for (let i = 0; i < file.ffProbeData.streams.length; i++) {
+    try {
+      if (
+        file.ffProbeData.streams[i].codec_type.toLowerCase() === 'subtitle'
+      ) {
+        hasSubs = true;
+      }
+    } catch (err) {}
+  }
+
+  if (hasSubs === false) {
+    response.infoLog += '☒ No subs in file, skipping!\n';
+    response.processFile = false;
+    return response;
+  } else {
+    response.infoLog += '☑ Found subs!\n';
+  }
+
+  // Set global variables.
+  const subsArr = file.ffProbeData.streams.filter((row) => row.codec_name === 'subrip');
+  const {
+    originalLibraryFile
+  } = otherArguments;
+  let extract_lang = inputs.extract_lang.split(',');
+  let subsFile = '';
+  let subsFile_forced = '';
+  let lang = '';
+  let title = '';
+  let forced = false;
+
+  let command = '-y,';
+  try {
```

Comment:
Please don't wrap the entire thing in a try/catch. This will silently fail causing Tdarr to think the plugin executed fine.

Reply:
Truth be told, I'm not proficient in .js, or any other language really. I've built this up bit by bit by studying code I've found around the web then trial and error to come up with something that works. If you could point me in a better direction to research a solution to handle this I'd appreciate it.